src/mpi4py/MPI.src/asbuffer.pxi

```diff
@@ -79,17 +91,28 @@ cdef extern from "Python.h":
     Py_ssize_t PyNumber_AsSsize_t(object, object) except? -1
 
 
+cdef inline int check_cpu_accessible(unsigned device_type) except -1:
+    if device_type == 0              : return 0
+    if device_type == kDLCPU         : return 0
+    if device_type == kDLOneAPI      : return 0  # ~> uncovered
```

Comment:
Based on my understanding in oneAPI (@oleksandr-pavlyk can correct me if I am wrong) it's not straightforward to use this like we do for CUDA (it's not a real pointer).

Reply:
The kDLOneAPI indicates that the pointer is a USM allocation make by oneAPI.  Whether MPI is able to natively send the content of USM allocation depends on the oneAPI backend that made the USM allocation. 

For USM device allocations on GPUs made by Level-Zero backend, Intel MPI is able to send and receive the buffer, and it would make sense to allow MPI to handle the transmission. Finding out the backend that made the allocation requires a call to oneAPI DPC++ runtime.

The USM allocations can also be host accessible (USM-host and USM-shared allocations can be accessed by Python interpreter), but recognizing these, again, requires interaction with DPC++ runtime. 

The safest assumption one can make without interacting with DPC++ runtime is that kDLOneAPI allocation is not CPU accessible.

Reply:
@oleksandr-pavlyk Thanks for your crystal-clear clarification. 

A minor off-topic comment, though: Shouldn't DLPack have a way to distinguish USM-host and USM-shared, for example `kDLOneAPIHost` and `kDLOneAPIShared`? Why should DLPack distinguish CUDA  but not oneAPI?

Reply:
Thanks, @oleksandr-pavlyk, for jumping in promptly!

> Shouldn't DLPack have a way to distinguish USM-host and USM-shared, for example `kDLOneAPIHost` and `kDLOneAPIShared`? Why should DLPack distinguish CUDA but not oneAPI?

Last time I heard from Sasha he was pondering on this question too (we also had a similar discussion). I dunno if he's figured out a solution?